api/submission/remove_submission.go

```diff
@@ -0,0 +1,44 @@
+package submission
+
+import (
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/db"
+)
+
+type RemoveSubmissionRequest struct {
+    core.APIRequestAssignmentContext
+    core.MinRoleGrader
+
+    TargetUser core.TargetUserSelfOrGrader `json:"target-email"`
+    TargetSubmission string `json:"target-submission"`
+}
+
+type RemoveSubmissionResponse struct {
+    FoundUser bool `json:"found-user"`
+    FoundSubmission bool `json:"found-submission"`
+}
+
+func HandleRemoveSubmission(request *RemoveSubmissionRequest) (*RemoveSubmissionResponse, *core.APIError){
+    response := RemoveSubmissionResponse{};
+
+    if (!request.TargetUser.Found) {
+        return &response, nil;
+    }
+
+    response.FoundUser = true;
+
+    doesExist, err := db.RemoveSubmission(request.Assignment, request.TargetUser.Email, request.TargetSubmission);
+    
+    if (doesExist) {
+        response.FoundSubmission = true;
+    } else { 
+        response.FoundSubmission = false;
+    }
```

Comment:
Just a nit, but you could just do:
```
response.FoundSubmission = doesExist;
```

If would make sense to use an `if` if there were other things you need to do for each case.